posts/2017-12-04-serverless-aurora-future-of-data.md

```diff
@@ -0,0 +1,122 @@
+<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+That's right. **Serverless Aurora.**
+
+Why is Serverless Aurora so important? We first need to understand two things: the technology-driven changes in software architectures in the cloud era, and the current state of the data layer in serverless architectures.
+
+# The Architectural Evolution
+
+Earlier this year, Adrian Cockcroft wrote a piece on the [Evolution of Business Logic from Monoliths through Microservices, to Functions](https://read.acloud.guru/evolution-of-business-logic-from-monoliths-through-microservices-to-functions-ff464b95a44d) that blew my mind. It showed how changes in technology are driving changes in development patterns and processes. Adrian has had a front row seat for these changes over the years from his work at eBay, Netflix, and now AWS.
+
+A bunch of unrelated technologies combined to drive these changes. Faster networks and better serialization protocols enabled compute that was distributed rather than centralized. This enabled API-driven architecture patterns that used managed services from SaaS providers and broke monoliths into microservices.
+
+Chef, Puppet, EC2 and Docker and eventually Lambda combined to enable and promote ephemeral compute environments that reduced time to value and increased utilization. These tools were combined with the necessary process improvements from the DevOps movement to increase velocity. We're seeing smaller teams deliver features faster with lower costs.
+
+These changes have been huge, but the data layer has been lagging. Adrian touched on database improvements, but they aren't as mind-blowing. They have explicit tradeoffs of simple query patterns:
+
+> Compared to relational databases, NoSQL databases provide simple but extremely cost effective, highly available and scalable databases with very low latency.
+
+The lagging data layer is particularly problematic in Serverless architectures.
+
+# The Problem of the Serverless Data Layer
+
+I [spoke on this problem](https://serverless.com/blog/serverless-conf-2017-nyc-recap/#data-layer-in-the-serverless-world) at ServerlessConf NYC in October. In short, there are two approaches you can take with databases with serverless compute: _server-full_ or _serverless_.
+
+### Server-full databases
+
+A server-full approach uses instance-based solutions such as MySQL, Postgres, or MongoDB. I classify them as instance-based when you can tell me how many instances you have running and what their hostnames are.
+
+I like Postgres + Mongo because of [their popularity](https://db-engines.com/en/ranking), which means data design patterns are well-known and language libraries are mature.
+
+However, these instance-based solutions were designed for a pre-serverless world with long-running compute instances. This leads to the following problems:
+
+_Connection Limits_
```

Comment:
IMHO the biggest problem for Lambda was establishing connection phase. It took some time to connect and it influenced the coldstart. The second issue that affected coldstart even more was that lambda had to run in VPC to connect to DB.

Reply:
In general I agree with everything you wrote in this section.